pegasus/src/social_metadata.rb

```diff
@@ -0,0 +1,160 @@
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "week-before-hoc" => {
+        title: I18n.t(:og_title_soon),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "actual-hoc" => {
+        title: I18n.t(:og_title_here),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      },
+      "default" => {
+        title: hoc_s(:meta_tag_og_title),
+        description: hoc_s(:meta_tag_og_description),
+        description_twitter: hoc_s(:meta_tag_twitter_description),
+        image: {path: "/images/social-media/hourofcode-2015-video-thumbnail.jpg", width: 1440, height: 900},
+        video: {youtube_key: "KsOIlDT145A", width: 640, height: 360}
+      }
+    },
+    "challenge" => {
+      "soon-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1920/social-media/hoc-student-challenge.jpg", width: 1920, height: 1080}
+      },
+      "week-before-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1920/social-media/hoc-student-challenge.jpg", width: 1920, height: 1080}
+      },
+      "actual-hoc" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      },
+      "default" => {
+        title: "Celebrity Challenge",
+        description: "Win a celebrity video chat for your class!",
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      }
+    }
+  }
+
+  if request.path == "/challenge" && request.site == "code.org"
+    page = "challenge"
+  elsif request.path == "/" && ["code.org", "csedweek.org", "hourofcode.com"].include?(request.site)
+    page = request.site
+  else
+    return {}
+  end
+
+  hoc_mode = DCDO.get("hoc_mode", "default")
+
+  # For now, post-hoc looks the same as the default.
+  if hoc_mode == "post-hoc"
+    hoc_mode = "default"
+  end
+
+  output = {}
+  social_tags[page][hoc_mode].each do |name, value|
+    if name == :image
```

Comment:
Is it possible for these to have multiple entries of the same name?  If not, maybe clearer to write this as:

```rb
if social_tags[page][hoc_mode][:image]
  # ...
elsif social_tags[page][hoc_mode][:video]
  # ...
```